module/data/item/templates/activated-effect.mjs

```diff
@@ -121,12 +170,39 @@ export default class ActivatedEffectTemplate extends SystemDataModel {
   }
 
   /* -------------------------------------------- */
-  /*  Migrations                                  */
+
+  /**
+   * Prepare a specific final formula, passing resolution errors to actor if available.
+   * @param {string} keyPath           Path within system data to where the property can be found.
+   * @param {object} options
+   * @param {string} options.label     Localizable name for the property to display in warnings.
+   * @param {object} options.rollData  Roll data to use to evaluate the formula.
+   */
+  _prepareFinalFormula(keyPath, { label, rollData }) {
+    const value = foundry.utils.getProperty(this, keyPath);
+    if ( !value ) return;
+    const property = game.i18n.localize(label);
+    try {
+      foundry.utils.setProperty(
+        this, keyPath, Roll.safeEval(replaceFormulaData(value, rollData, { actor: this.parent.actor, property }))
+      );
+    } catch(err) {
+      if ( this.parent.isEmbedded ) {
+        const message = game.i18n.format("DND5E.FormulaMalformedError", { property, name: this.parent.name });
+        this.parent.actor._preparationWarnings.push({ message, link: this.parent.uuid, type: "error" });
+        console.error(message, err);
+      }
+    }
+  }
+
+  /* -------------------------------------------- */
+  /*  Data Migration                              */
   /* -------------------------------------------- */
 
   /** @inheritdoc */
   static _migrateData(source) {
     super._migrateData(source);
+    if ( ("activation" in source) && !source.activation.type ) source.activation.type = "";
```

Comment:
If you update only part of the activation, this will clear the type:

```js
await item.update({ "system.activation": { cost: 1, type: "special" } });
console.log(item.system.activation.type); // "special"
await item.update({ "system.activation.cost": 2 });
console.log(item.system.activation.type); // ""
```